src/Core/src/Handlers/ShapeView/ShapeViewHandler.Android.cs

```diff
@@ -19,9 +19,9 @@ public static void MapBackground(IShapeViewHandler handler, IShapeView shapeView
 			{
 				handler.UpdateValue(nameof(IViewHandler.ContainerView));
 				handler.ToPlatform().UpdateBackground(shapeView);
-
-				handler.PlatformView?.InvalidateShape(shapeView);
 			}
+
+			handler.PlatformView?.InvalidateShape(shapeView);
```

Comment:
There are several properties for defining the background color in Views. Specifically in Shapes we have Background, BackgroundColor and Fill. Fill is the specific property in Shapes to fill it. If use Fill, Background or BackgroundColor must be used in the container. For example, if we have an Ellipse with Red Fill and Yellow Background, it should render a red circle (Ellipse) on a yellow rectangle (ContainerView).

Applying changes to BackgroundColor, Background:
`public static void MapBackground(IShapeViewHandler handler, IShapeView shapeView)`

So, the previous condition is not taking into account the BackgroundColor, should be something like:
`if ((shapeView.BackgroundColor is not null || shapeView.Background is not null) && shapeView.Fill is not null)`

Reply:
But in this case background is not null

```c#
Paint? IView.Background
{
	get
	{
		if (!Brush.IsNullOrEmpty(Background))
			return Background;
		if (BackgroundColor.IsNotDefault())
			return new SolidColorBrush(BackgroundColor);

		return null;
	}
}
``

It just doesn't invalidate changes because shapeView.Fill is null

Reply:
Fine, could you apply the changes?

Reply:
Done. Could you please /azp?